06_puppy/kasunfdo/store_test.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type StorerTest struct {
+	suite.Suite
+	store Storer
+	id    uint64
+}
+
+func (suite *StorerTest) SetupTest() {
+	suite.id = suite.store.CreatePuppy(Puppy{Breed: "Labrador", Colour: "Cream", Value: 2999.99})
+}
+
+func (suite *StorerTest) TestCreatePuppy() {
+	id := suite.store.CreatePuppy(Puppy{Breed: "German Shepard", Colour: "Brown", Value: 3499.99})
+	suite.True(id > 1)
+}
+
+func (suite *StorerTest) TestReadPuppy() {
+	puppy, exists := suite.store.ReadPuppy(suite.id)
+
+	suite.True(exists)
+	suite.Equal(puppy.ID, suite.id)
+	suite.Equal(puppy.Breed, "Labrador")
+	suite.Equal(puppy.Colour, "Cream")
+	suite.Equal(puppy.Value, 2999.99)
+
+	_, exists = suite.store.ReadPuppy(100)
+	suite.False(exists)
+}
+
+func (suite *StorerTest) TestUpdatePuppy() {
+	isSuccessful := suite.store.UpdatePuppy(suite.id, Puppy{Breed: "Labrador Retriever", Colour: "Brown", Value: 3999.99})
+
+	suite.True(isSuccessful)
+	puppy, isSuccessful := suite.store.ReadPuppy(suite.id)
+
+	suite.True(isSuccessful)
+	suite.Equal(puppy.ID, suite.id)
+	suite.Equal(puppy.Breed, "Labrador Retriever")
+	suite.Equal(puppy.Colour, "Brown")
+	suite.Equal(puppy.Value, 3999.99)
+
+	isSuccessful = suite.store.UpdatePuppy(100, Puppy{Breed: "Poodle", Colour: "White", Value: 1999.99})
+	suite.False(isSuccessful)
+}
+
+func (suite *StorerTest) TestDeletePuppy() {
+	isSuccessful := suite.store.DeletePuppy(suite.id)
+	suite.True(isSuccessful)
+
+	_, isSuccessful = suite.store.ReadPuppy(suite.id)
+	suite.False(isSuccessful)
+
+	isSuccessful = suite.store.DeletePuppy(suite.id)
+	suite.False(isSuccessful)
+}
+
+func TestMapStore(t *testing.T) {
+	s := StorerTest{
+		store: NewMapStore(),
+	}
+	suite.Run(t, &s)
```

Comment:
IMO you should just fold this into one line (fee free to argue against it):
```go
suite.Run(t, &StorerTest{store: NewMapStore()})
```
same below.

Reply:
Done